eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/CollectionResultDefinition.cs

```diff
@@ -4,46 +4,125 @@
+        private readonly InputResponseLocation? _nextPageLocation;
+        private readonly string _getNextResponseMethodName;
 
-        private static readonly ParameterProvider ContinuationTokenParameter =
-            new("continuationToken", $"The continuation token.", new CSharpType(typeof(string)))
-            {
-                DefaultValue = Null
-            };
+        private readonly IReadOnlyList<ParameterProvider> _createRequestParameters;
+        private readonly FieldProvider? _contextField;
+
+        private static readonly ParameterProvider NextLinkParameter =
+            new("continuationToken", $"The continuation token.", new CSharpType(typeof(string), isNullable: true));
 
         private static readonly ParameterProvider PageSizeHintParameter =
-            new("pageSizeHint", $"The page size hint.", new CSharpType(typeof(int?)))
-            {
-                DefaultValue = Null
-            };
+            new("pageSizeHint", $"The page size hint.", new CSharpType(typeof(int?)));
 
         public CollectionResultDefinition(ClientProvider client, InputPagingServiceMethod serviceMethod, CSharpType? itemModelType, bool isAsync)
         {
             _client = client;
+            _paging = serviceMethod.PagingMetadata;
+            _getNextResponseMethodName = _isAsync ? "GetNextResponseAsync" : "GetNextResponse";
+
+            var nextPagePropertyName = _paging.NextLink != null
+                ? _paging.NextLink.ResponseSegments[0]
+                : _paging.ContinuationToken?.ResponseSegments[0];
+            _nextPageLocation = _paging.NextLink?.ResponseLocation ?? _paging.ContinuationToken?.ResponseLocation;
+            _clientField = new FieldProvider(
+                FieldModifiers.Private | FieldModifiers.ReadOnly,
+                _client.Type,
+                "_client",
+                this);
             _operation = serviceMethod.Operation;
             _itemModelType = itemModelType ?? new CSharpType(typeof(BinaryData));
             _isAsync = isAsync;
+            _createRequestParameters = _client.RestClient.GetCreateRequestMethod(_operation).Signature.Parameters;
+            var fields = new List<FieldProvider>();
+            for (int paramIndex = 0; paramIndex < _createRequestParameters.Count; paramIndex++)
+            {
+                var parameter = _createRequestParameters[paramIndex];
+                var field = new FieldProvider(
+                    FieldModifiers.Private | FieldModifiers.ReadOnly,
+                    parameter.Type,
+                    $"_{parameter.Name}",
+                    this);
+                fields.Add(field);
+                if (field.Name == "_context")
+                {
+                    _contextField = field;
+                }
+            }
+
+            _requestFields = fields;
+            _scopeName = $"{_client.Name}.{_operation.Name}"; // TODO - may need to expose ToCleanName for the operation
```

Comment:
pending on https://github.com/microsoft/typespec/issues/7386

Reply:
I think we can update CreateClientCollectionResultDefinition to take the MethodProvider.

Reply:
will handle this in later PRs

Reply:
As offline discussed, expose ToCleanIdentifierName from MTG